### PR TITLE
Pull context up in a couple places

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: clients/redshift/merge.go [hunk 1]
```diff
@@ -83,7 +84,7 @@ func (s *Store) Merge(ctx context.Context, tableData *optimization.TableData) er
 			continue
 		}
 
-		err = utils.BackfillColumn(ctx, s, col, fqName)
+		err = utils.BackfillColumn(*config.FromContext(ctx).Config, s, col, fqName)
 		if err != nil {
```
Comment (Tang8330) [inline clients/redshift/merge.go:88]: Can we inject the Store object with `config` and just pass it around when necessary?

Comment (nathan-artie) [reply clients/redshift/merge.go:88]: You mean pass store to `utils.BackfillColumn`?

Comment (nathan-artie) [reply clients/redshift/merge.go:88]: And also `TableData`?

Comment (Tang8330) [reply clients/redshift/merge.go:88]: Sorry for not being clear.

I meant have `config` as an object on store, such that this function becomes something like:

```go
err = utils.BackfillColumn(s.Config, s, col, fqName)
```

Comment (nathan-artie) [reply clients/redshift/merge.go:88]: Yep, will do that in a follow up.

**Pull request merged**